### PR TITLE
Remove data attributes from being saved in html of All products block

### LINKED ISSUE #9961: All Products Block saving data attributes in HTML
The `All Products` block currently saves the attributes data into the HTML seen here https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/blocks/products/all-products/save.js#L27

Since these attributes data is already handled by Gutenberg we don't need to do that. This issue is to remove those as seen done in https://github.com/woocommerce/woocommerce-blocks/pull/9833

### PR DESCRIPTION
**Pull request by hritikchaudhary**
Fixes #9961 

### Changes in the PR:
Removed propTypes definitions from Reviews By Category.

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [x] E2E tests

#### User Facing Testing

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

**Comment by roykho** [comment]
Hi @hritikchaudhary - thanks for submitting this however you will need to deprecate here as we discussed on https://github.com/woocommerce/woocommerce-blocks/issues/9961 otherwise users will see block validation errors.

**Comment by roykho** [comment]
Hi @hritikchaudhary - any updates on this? If you don't have time to look more into this, I am more than happy to take this over.

**Comment by hritikchaudhary** [comment]
Hi @roykho I'm really sorry, this one slipped out of my radar...
I saw there is already a [deprecated.js](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/blocks/products/all-products/deprecated.js) file in the same directory, do I have to use that or create a new one?

**Comment by roykho** [comment]
Hi @hritikchaudhary - yes you can use the same file and just create a v2 object.

**Comment by hritikchaudhary** [comment]
Ok, Thanks for the clarification.

**Comment by hritikchaudhary** [comment]
Hi @roykho, I have added old save function in [deprecated.js](https://github.com/woocommerce/woocommerce-blocks/pull/10032/files#diff-20bdccaaf2cad73ed7adeae966276a4c5867c02b542bf16e6cebef31416b2b27), Not sure how to modify [save.js](https://github.com/woocommerce/woocommerce-blocks/pull/10032/files#diff-63c8840763e14662950356f92ede146e565e5bc57d99fb0f2ca6923109ee346d) so it indicates that save function is deprecated.
Apologies, I'm still learning react, I am more experienced with Angular 😬

File: assets/js/blocks/products/all-products/save.js [hunk 1]
```diff
@@ -15,16 +15,12 @@ export default function save( { attributes } ) {
 		.forEach( ( key ) => {
 			dataAttributes[ key ] = attributes[ key ];
 		} );
```
Comment (roykho) [inline assets/js/blocks/products/all-products/save.js]: Pretty sure line 12-17 is no longer needed right?

File: assets/js/blocks/products/all-products/deprecated.js [hunk 2]
```diff
@@ -33,4 +33,29 @@ const v1 = {
 	},
 };
 
-export default [ v1 ];
+const v2 = {
+	save( { attributes } ) {
```
Comment (roykho) [inline assets/js/blocks/products/all-products/deprecated.js:40]: Pretty sure we need to add the attributes object here just like in v1

File: assets/js/blocks/products/all-products/deprecated.js [hunk 3]
```diff
@@ -33,4 +33,29 @@ const v1 = {
...
+				) }
+				{ ...data }
+			>
+				<InnerBlocks.Content />
+			</div>
+		);
+	},
+};
+
+export default [ v1, v2 ];
```
Comment (roykho) [inline assets/js/blocks/products/all-products/deprecated.js]: v2 needs to come before v1

**Review by roykho — CHANGES_REQUESTED** [review]
Thanks for pushing up updates. I've left some comments.

File: assets/js/blocks/products/all-products/save.js (hunk 1 above)
Comment (hritikchaudhary) [reply assets/js/blocks/products/all-products/save.js]: removed line 12-17.

File: assets/js/blocks/products/all-products/deprecated.js (hunk 2 above)
Comment (hritikchaudhary) [reply assets/js/blocks/products/all-products/deprecated.js:40]: Updated v2 with attributes object just like in v1

File: assets/js/blocks/products/all-products/deprecated.js (hunk 3 above)
Comment (hritikchaudhary) [reply assets/js/blocks/products/all-products/deprecated.js]: updated

**Review by roykho — CHANGES_REQUESTED** [review]
Hi @hritikchaudhary - some e2e tests are not passing as [seen here](https://github.com/woocommerce/woocommerce-blocks/pull/10444) could you please take a look at those and fix/update the tests if needed so that they will pass?

**Comment by hritikchaudhary** [comment]
Hi @roykho I'm facing some discrepancies in e2e test result, but was able to pin point one test case that was common between my local and workflow result.
[blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:122:7 › woocommerce/mini-cart Block › with All products Block › should open the filled cart drawer 
Can you please provide some guidance related to how to fix this e2e testcase... or just point where I can look for help and debug, that'd be very helpful.
Thanks!

**Pull request merged**